build/__main__.py

```diff
@@ -27,33 +29,31 @@ def _error(msg, code=1):  # type: (str, int) -> None  # pragma: no cover
     exit(code)
 
 
-def build(srcdir, outdir, distributions, skip_dependencies=False):  # type: (str, str, List[str], bool) -> None
-    '''
-    Runs the build process
-
-    :param srcdir: Source directory
-    :param outdir: Output directory
-    :param distributions: Distributions to build (sdist and/or wheel)
-    :param skip_dependencies: Do not perform the dependency check
-    '''
-    try:
-        builder = ProjectBuilder(srcdir)
-
-        for dist in distributions:
-            if not skip_dependencies:
-                missing = builder.check_depencencies(dist)
-                if missing:
-                    _error('Missing dependencies:' + ''.join(['\n\t' + dep for dep in missing]))
-
-            builder.build(dist, outdir)
-    except BuildException as e:
-        _error(str(e))
-    except BuildBackendException as e:
-        if sys.version_info >= (3, 5):  # pragma: no cover
-            print(traceback.format_exc(-1))
-        else:  # pragma: no cover
-            print(traceback.format_exc())
-        _error(str(e))
+class MainBuilder(ProjectBuilder):
+    def build_distributions(self, outdir, distributions, skip_dependencies=False):  # type: (str, List[str], bool) -> None
+        '''
+        Runs the build process
+
+        :param outdir: Output directory
+        :param distributions: Distributions to build (sdist and/or wheel)
+        :param skip_dependencies: Do not perform the dependency check
+        '''
+        try:
+            for dist in distributions:
+                if not skip_dependencies:
+                    missing = self.check_depencencies(dist)
+                    if missing:
+                        _error('Missing dependencies:' + ''.join(['\n\t' + dep for dep in missing]))
```

Comment:
These checks seem redundant for the isolated builds 🤔 They can only happen with no isolation, not?